### PR TITLE
Support request notification 207

### PR DESCRIPTION
**Pull request by kyaroch**
## Changelog
- Creates a note with creator's name on support request creation
- Sends an email to creator and lockbox partner users on support request creation
- Makes `user` optional for notes (restoring the possibility of system-created notes)

## Link to issue:  
Fixes issue #207 

## Relevant Screenshots: 

Note:

Screenshot_20191013_182543

Example email:
```
Date: Sun, 13 Oct 2019 18:20:08 -0500
From: from@example.com
To: yawny@sloths.com
Message-ID: <5da3b12870071_2ce2ab47a770b1053ce@kevin-asus.mail>
Subject: A new Support Request was sent to Healthy Sloths
Mime-Version: 1.0
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

A new support request was sent to Healthy Sloths by Sally Snake.

Pickup date: Monday, October 21, 2019
Client alias: oijf
Client reference ID: woeifkm
Urgency flag: aaaa
Amount: $27.00
Amount breakdown:
  - $12.00 for gas
  - $15.00 for parking
```

## Are you ready for review?:

- [ x] Added relevant tests
- [ x] Linked PR to the issue
- [ x] Added notes for QA/special notes
- [ x] Added revelant screenshots

### CODE REVIEW

File: app/views/support_request_mailer/creation_alert.text.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Pickup date: <%= @support_request.pickup_date.strftime("%A, %B %d, %Y") %>
+Client alias: <%= @support_request.name_or_alias %>
+Client reference ID: <%= @support_request.client_ref_id %>
+<% if @support_request.urgency_flag.present? %>
+Urgency flag: <%= @support_request.urgency_flag %>
+<% end %>
+Amount: <%= @support_request.amount.format %>
+Amount breakdown:
+<% @support_request.lockbox_transactions.each do |transaction| %>
+  - <%= transaction.amount.format %> for <%= transaction.category %>
```
Comment (BintLopez) [inline app/views/support_request_mailer/creation_alert.text.erb]: let's not show this to the partner

File: app/views/support_request_mailer/creation_alert.text.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+A new support request was sent to <%= @support_request.lockbox_partner.name %> by <%= @support_request.user.name %>.
+
+Pickup date: <%= @support_request.pickup_date.strftime("%A, %B %d, %Y") %>
+Client alias: <%= @support_request.name_or_alias %>
```
Comment (BintLopez) [inline app/views/support_request_mailer/creation_alert.text.erb]: let's drop the name

File: app/views/support_request_mailer/creation_alert.text.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+Client reference ID: <%= @support_request.client_ref_id %>
```
Comment (BintLopez) [inline app/views/support_request_mailer/creation_alert.text.erb]: only show this if it exists

**Comment by kyaroch** [comment]
Here is the new HTML email. (Note that I just rendered the HTML directly in a browser; most email clients will make the URLs clickable automatically.)

Screenshot_20191020_181354

File: app/mailers/support_request_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    urgency_flag_prefix = if @support_request.urgency_flag.present?
+      "#{@support_request.urgency_flag} - "
+    else
+      ""
+    end
+
+    subject = "#{urgency_flag_prefix}MAC Cash Box Withdrawal Request"
+
+    mail(to: email_addresses, subject: subject, bcc: @support_request.user.email)
```
Comment (BintLopez) [inline app/mailers/support_request_mailer.rb]: let's add lockbox@midwestaccesscoalition.org to the bcc list

Comment (BintLopez) [inline app/mailers/support_request_mailer.rb]: we should add some other email addys to the BCC... will discuss in Slack

File: app/models/support_request.rb [hunk 5]
```diff
@@ -49,6 +49,14 @@ def status_options
     LockboxAction::STATUSES - [status]
   end
 
+  def record_creation
+    note_text =
+      "Support request generated by #{user.name} " \
+      "at #{created_at.strftime("%I:%M %p %:::z")} " \
+      "on #{created_at.strftime("%B %d, %Y")}"
+    notes.create(text: note_text)
```
Comment (BintLopez) [inline app/models/support_request.rb]: 💯

File: app/mailers/support_request_mailer.rb (hunk 4 above)
Comment (BintLopez) [inline app/mailers/support_request_mailer.rb]: let's CC the coordinator

File: lib/create_support_request.rb [hunk 6]
```diff
@@ -51,6 +52,8 @@ def call
       end
     end
 
+    support_request.record_creation
+    send_creation_alert
```
Comment (BintLopez) [inline lib/create_support_request.rb]: what happens if either of these error out?

File: spec/lib/create_support_request_spec.rb [hunk 7]
```diff
@@ -61,7 +88,10 @@
     it 'goes to the finance team when balance is below $300' do
       ENV['LOW_BALANCE_ALERT_EMAIL'] ||= 'lowbalance@alert.com'
 
-      result = CreateSupportRequest.call(params: low_balance_params)
+      result = nil
+
+      expect { result = CreateSupportRequest.call(params: low_balance_params) }
+        .to change{ActionMailer::Base.deliveries.length}.by(2)
```
Comment (BintLopez) [inline spec/lib/create_support_request_spec.rb]: 🙌

**Pull request merged**